src/tests/domain-test/user-subscription.test.ts

```diff
@@ -12,26 +15,52 @@ describe("Subscribe user", () => {
     expect(await userSubscription.subscribeUser(user)).to.be.true;
   });
 
-  it("Should return error when subscribing the user fails", async () => {
-    const user = new User();
+  it("Should save user id", async () => {
+    let user = new User();
     user.firstName = "Rorschasch";
-    user.id = 123;
-    try {
-      return await userSubscription.subscribeUser(user);
-    } catch (error) {
-      expect(error.message).to.be.eq("Failed to subscribe user.");
-    }
+    user.id = 123445;
+    await userSubscription.subscribeUser(user);
+    let saved = await userDatasource.findUserById(user.id);
+    expect(saved!.id).to.be.eq(user.id);
+  });
+
+  it("Should save user firstName", async () => {
+    let user = new User();
+    user.firstName = "Rorschasch";
+    user.id = 123445;
+    await userSubscription.subscribeUser(user);
+    let saved = await userDatasource.findUserById(user.id);
+    expect(saved!.firstName).to.be.eq(user.firstName);
+  });
+
+  it("Should save user as active true when subscribing", async () => {
+    let user = new User();
+    user.firstName = "Rorschasch";
+    user.id = 123445;
+    await userSubscription.subscribeUser(user);
+    let saved = await userDatasource.findUserById(user.id);
+    expect(saved!.isActive).to.be.true;
+  });
+
+  it("Should save user as active false when unsubscribing", async () => {
+    let user = new User();
+    user.firstName = "Rorschasch";
+    user.id = 123445;
+    await userSubscription.subscribeUser(user);
+    await userSubscription.unsubscribeUser(user);
+    let saved = await userDatasource.findUserById(user.id);
+    expect(saved!.isActive).to.be.false;
```

Comment:
Aqui eu vou te recomendar um padrão simples para evitar esse monte de código junto: 

https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80